snuba/request/exceptions.py

```diff
@@ -0,0 +1,7 @@
+class InvalidJsonRequestException(Exception):
```

Comment:
I did not put this exception in query/parser/exceptions.py because it represents errors happening in a different phase of the query and I would not want to couple them.

This means "I could not parse the json payload and I could not build a meaningful request". Which is dependent on the transport being json. 

InvalidQueryException (in query/parser) instead means: "I could not build a meaningful query out of a valid payload". That parsing phase should not be dependent on the fact that the payload is json. Thus the entire parser module should not depend on this exception.

Reply:
I don't have a very strong opinion of this but I think it might make sense to consider "the body was not valid JSON" and "the JSON does not conform to the expected schema" as two distinct classes of errors.